be/src/exprs/vectorized/cast_expr_array.cpp

```diff
@@ -73,6 +77,99 @@ StatusOr<ColumnPtr> VectorizedCastArrayExpr::evaluate_checked(ExprContext* conte
     return cast_column;
 };
 
+inline bool is_quote(char c) {
+    return c == '\'' || c == '"';
+}
+
+bool is_valid_array(const Slice& src, std::vector<char>& container) {
+    size_t length = src.get_size();
+    container.clear();
+
+    for (size_t i = 0; i < length; ++i) {
+        // process string literal
+        if (!container.empty() && is_quote(container.back())) {
+            if (container.back() != src[i]) {
+                continue;
+            }
+        }
+        if (is_quote(src[i])) {
+            if (!container.empty() && container.back() == src[i]) {
+                container.pop_back();
+            } else {
+                container.push_back(src[i]);
+            }
+        } else if (src[i] == '[') {
+            container.push_back(src[i]);
+        } else if (src[i] == ']') {
+            if (!container.empty() && container.back() == '[') {
+                container.pop_back();
+            } else {
+                return false;
+            }
+        }
+    }
+
+    return container.empty();
+}
+
+// [[],[]] -> [],[]
+// [],[] -> [],[]
+// [[]] -> []
+Slice strip_array_wrapper(const Slice& src) {
+    size_t length = src.get_size();
+    if (length < 2) {
+        return src;
+    }
+
+    size_t stack_size = src[0] == '[';
+
+    for (size_t i = 1; i < length - 1; ++i) {
+        stack_size += src[i] == '[';
+        stack_size -= src[i] == ']';
+
+        if (stack_size == 0) {
+            return src;
+        }
+    }
+
+    return {src.get_data() + 1, src.get_size() - 2};
+}
+
+//
+void array_delimeter_split(const Slice& src, std::vector<Slice>& res, std::vector<char>& stack) {
```

Comment:
Stupid question. How to generate multidimensional from this function if `res` is std::vector? add some docs?